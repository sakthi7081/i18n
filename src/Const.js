export const langauges = ["en", "ta", "te", "hi", "ml", "it", "ja"];

export const color = "#0ecea6";
