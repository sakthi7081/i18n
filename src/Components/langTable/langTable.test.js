import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { findTestAttr } from "./../../Utility/Utility";
import LangTable from "./langTable";
import i18n from "../../i18n";
import { langauges } from "../../Const";

langauges.map((lan, i) =>
  i === 2
    ? ""
    : i18n.addResourceBundle(
        lan,
        "translation",
        i === 1 ? { Value: "value" } : { Data: "Data" },
        true,
        true
      )
);

const setUp = () => {
  const component = shallow(<LangTable />);
  return component.dive().dive();
};

const restProps = {
  tableRow: { row: "row" },
  tableColumn: { column: { name: "a" } },
  style: {},
};

describe("Table Component", () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });

  it("Should render the component with no error", () => {
    const wrapper = findTestAttr(component, "Table-Component");
    expect(wrapper.length).toBe(1);
  });

  it("Should render the HeaderRow", () => {
    const wrapper = findTestAttr(component, "Header");
    wrapper.props().rowComponent(restProps);
    expect(wrapper.length).toBe(1);
  });

  it("Should render the Headercell", () => {
    const wrapper = findTestAttr(component, "Header");
    wrapper.props().cellComponent(restProps);
    expect(wrapper.length).toBe(1);
  });

  it("Should render the fixedcolumn", () => {
    const wrapper = findTestAttr(component, "Fixed-Column");
    wrapper.props().cellComponent(restProps);
    expect(wrapper.length).toBe(1);
  });

  it("Should render the fixedcolumn Header", () => {
    const props = {
      tableRow: { rowId: "row" },
      tableColumn: { column: { name: "a" } },
      style: {},
    };
    const wrapper = findTestAttr(component, "Fixed-Column");
    wrapper.props().cellComponent(props);
    expect(wrapper.length).toBe(1);
  });

  it("Should match the snapShot", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
