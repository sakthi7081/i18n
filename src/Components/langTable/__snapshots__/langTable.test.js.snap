// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Component Should match the snapShot 1`] = `
<WithStyles(ForwardRef(Paper))
  className="sc-AxhCb jAkNLU"
  data-test="Table-Component"
  elevation={6}
>
  <Grid
    columns={
      Array [
        Object {
          "name": "key",
          "title": "key",
        },
        Object {
          "name": "en",
          "title": "en",
        },
        Object {
          "name": "ta",
          "title": "ta",
        },
        Object {
          "name": "te",
          "title": "te",
        },
        Object {
          "name": "hi",
          "title": "hi",
        },
        Object {
          "name": "ml",
          "title": "ml",
        },
        Object {
          "name": "it",
          "title": "it",
        },
        Object {
          "name": "ja",
          "title": "ja",
        },
      ]
    }
    data-test="Grid"
    rows={
      Array [
        Object {
          "en": "Data",
          "hi": "Data",
          "it": "Data",
          "ja": "Data",
          "key": "Data",
          "ml": "Data",
          "ta": "",
          "te": "",
        },
        Object {
          "en": "",
          "hi": "",
          "it": "",
          "ja": "",
          "key": "Value",
          "ml": "",
          "ta": "value",
          "te": "",
        },
      ]
    }
  >
    <Component />
    <SortingStateBase
      columnSortingEnabled={true}
      defaultSorting={Array []}
    />
    <IntegratedSortingBase />
    <SearchStateBase
      defaultValue=""
    />
    <IntegratedFilteringBase />
    <Component />
    <Component
      defaultOrder={
        Array [
          "key",
          "en",
          "ta",
          "te",
          "hi",
          "ml",
          "it",
          "ja",
        ]
      }
    />
    <Component
      cellComponent={[Function]}
      data-test="Header"
      rowComponent={[Function]}
      showSortingControls={true}
    />
    <Component />
    <Component />
    <Component
      cellComponent={[Function]}
      data-test="Fixed-Column"
      leftColumns={
        Array [
          "key",
        ]
      }
    />
  </Grid>
</WithStyles(ForwardRef(Paper))>
`;
